assertj-core/src/main/java/org/assertj/core/api/AssertFactory.java

```diff
@@ -12,6 +12,10 @@
  */
 package org.assertj.core.api;
 
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.function.Function;
+
 /**
  * A single method factory interface to create an {@link Assert} for a given value.
```

Comment:
This should be improved as it's no longer a single-method factory interface. In addition, adding some usage examples for the two `createAssert` methods would be beneficial.